src/crypto/crypto_hash.cc

```diff
@@ -94,7 +94,26 @@ void Hash::New(const FunctionCallbackInfo<Value>& args) {
     md = EVP_MD_CTX_md(orig->mdctx_.get());
   } else {
     const Utf8Value hash_type(env->isolate(), args[0]);
```

Comment:
This decoding in C++ land can be avoided if we map the hash to an ID in JS land and use that ID to index into an array of EVP_MD pointers.